BUILD.bazel

```diff
@@ -71,16 +71,28 @@ sonatype_artifact_bundle(
 
 jar_jar(
     name = "compiler_shaded",
-    input_jar = ":compiler_unshaded_deploy.jar",
+    input_jar = ":compiler_uberjar_deploy.jar",
     rules = "rules.jarjar",
 )
 
 java_binary(
-    name = "compiler_unshaded",
+    name = "compiler_uberjar",
     main_class = "com.google.javascript.jscomp.CommandLineRunner",
     runtime_deps = [":compiler_lib"],
 )
 
+# Produce an unshaded export jar based on :compiler_lib
+load("@rules_jvm_external//:defs.bzl", "java_export")
+version = "$(COMPILER_VERSION)" or "1.0-SNAPSHOT"
```

Comment:
This was added for completeness, using the same variable name and default value as was found in `bazel/sonatype_artifact_bundle.bzl`. Strictly speaking, these coords have no effect, as long as only one java_export rule exists in the entire project.

Reply:
You say "for completeness", but isn't this the jar that you wanted when you created this PR?
Isn't this the one that is expected to have only closure-compiler code and leave all of its dependencies out?

If not, which one is?

Reply:
Ah sorry for lack of clarity on the self-review content: the "completeness" here was the redundant version that is also found in `bazel/sonatype_artifact_bundle.bzl`. 

From what I can tell, in a "do everything the way that bazelbuild/rules_jvm_external would expect", this java_export (and matching ones for other artifacts) would be the way to create maven-consumable artifacts, and the sonatype_artifact_bazel file (and the .github/ci_support/deploy_sonatype_snapshot_bundles.js file) would probably go away or become a lot less complex.

I don't like duplicated logic (and constants), but couldn't find a clearer way to express this in bazel such that the version was shared between the two places.